### PR TITLE
[VIS] Grey out the role if IIGO is not run and additional info about IIGO status.

### PR DESCRIPTION
**Pull request by ramon15749**
# Summary

- Make the role greyed out if IIGO is not run at all (i.e. quit during broadcastTax). 
- Add the status table to account for turns that IIGO is run partially. 

ps. look mom (@preetl ) no hands <3

## Additional Information

image

## Test Plan

🤞

### CODE REVIEW

File: website/src/components/Visualisations/Roles/Util/ProcessedRoleData.ts [hunk 1]
```diff
@@ -48,34 +47,41 @@ export const processRoleData = (data: OutputJSONType): ProcessedRoleData => {
...
-                            elem.occupied,
-                            gameState.SpeakerID
-                        )
-                        break
+            elem.occupied = data.GameStates.slice(0, -1).reduce(
+                (acc, gameState, index) => {
+                    // taking index+1 because the IIGO status is only reported the next turn
+                    const DidntRun = data.GameStates[
+                        index + 1
+                    ].IIGORunStatus.includes('broadcastTaxation')
```
Comment (lhl2617) [inline website/src/components/Visualisations/Roles/Util/ProcessedRoleData.ts:55]: very hacky.


I like.

File: website/src/components/Visualisations/Roles/Util/RoleTypes.ts [hunk 2]
```diff
@@ -181,3 +201,10 @@ export type ProcessedRoleElem = {
 }
 
 export type ProcessedRoleData = ProcessedRoleElem[]
+
+export type IIGOStatus = {
+    turn: number
+    status: string
```
Comment (lhl2617) [inline website/src/components/Visualisations/Roles/Util/RoleTypes.ts:207]: there is a component name IIGOStatus already

File: website/src/components/Visualisations/Roles/Util/getIIGOStatuses.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const retData: IIGOStatuses = []
+
+    return data.GameStates.slice(1).map<IIGOStatus>((gameState) => {
+        const info: IIGOStatus = {
+            turn: gameState.Turn - 1,
+            status: gameState.IIGORunStatus,
+        }
+
+        return info
+    }, retData)
```
Comment (lhl2617) [inline website/src/components/Visualisations/Roles/Util/getIIGOStatuses.ts]:
```suggestion
    return data.GameStates.slice(1).map((gameState) => {
        return {
            turn: gameState.Turn - 1,
            status: gameState.IIGORunStatus,
        }
    })
```

Comment (lhl2617) [reply website/src/components/Visualisations/Roles/Util/getIIGOStatuses.ts]: does this work?

**Review by lhl2617 — APPROVED** [review]
preapproving :) just smol nits

File: website/src/components/Visualisations/Roles/Util/getIIGOStatuses.ts (hunk 3 above)
Comment (ramon15749) [reply website/src/components/Visualisations/Roles/Util/getIIGOStatuses.ts]: yup, it works! Thank you 😄

**Comment by preetl** [comment]
YOU'RE MY HERO THANK YOUUU <3

**Pull request merged**